### DISCUSSION TITLE
(waveup팀)아두이노 환경에서 Euphony Library

### DISCUSSION THREAD

**Discussion by Han-D-Peter** [euphony-io/euphony#151] [Q&A]
# 아두이노 환경에서 Euphony
아두이노를 통해 정해진 데이터를 스피커로 송출할 수 있는 방법을 찾고 있습니다.
[euphony](https://github.com/euphony-io/euphony/search?l=c%2B%2B) 여기에 있는 헤더파일들을 아두이노 환경에서 사용해도 괜찮을까요?
@sunyeongan @defwdahyun0

**Comment by designe** [comment]
안녕하세요! 
euphony 프로젝트의 아두이노 포팅을 말씀하시는거죠?
euphony 프로젝트는 하드웨어 (스피커, 마이크) dependency 때문에 바로 이기종 플랫폼에 포팅하는게 쉽지 않습니다.
현재는 안드로이드와 웹만 지원합니다.

아두이노에 포팅하기 위해서는 코드내에서 스피커와 마이크 API를 직접 사용하는 부분에서 아두이노에 맞게 코드 작성이 필요합니다. 
안드로이드에서는 마이크, 스피커 접근을 위해 AAudio와 Android API에서 제공하는 API를 사용하며, Web에서는 Web Audio API를 사용합니다.

물론 음파 생성되는 부분의 클래스만 가져다가 쓰고 싶으신 경우에는 오픈소스 라이센스 명시만 하시면 언제든지 쓰셔도 됩니다 :)
